extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/VertxRequestHandler.java

```diff
@@ -74,18 +84,57 @@ public VertxRequestHandler(Vertx vertx,
         this.currentVertxRequest = CDI.current().select(CurrentVertxRequest.class).get();
     }
 
+    private static Uni<MessageReader> messageReader(RoutingContext routingContext) {
+        return Uni.createFrom().emitter(uniEmitter -> {
+            try {
+                final HttpServerRequest request = routingContext.request();
+                final Consumer<BiConsumer<String, String>> forEachHeader = processHeader -> {
+                    request.headers().forEach(kvp -> processHeader.accept(kvp.getKey(), kvp.getValue()));
+                };
+
+                request.bodyHandler(body -> {
+                    uniEmitter.complete(HttpMessageFactory.createReader(forEachHeader, body.getBytes()));
+                });
+            } catch (Throwable t) {
+                uniEmitter.fail(t);
+            }
+        });
+    }
+
+    private static MessageWriter messageWriter(RoutingContext routingContext) {
+        final HttpServerResponse response = routingContext.response();
+        final BiConsumer<String, String> putHeader = (k, v) -> response.putHeader(k, v);
+
+        return HttpMessageFactory.createWriter(putHeader, body -> {
+            if (body != null && body.length > 0) {
+                response.end(Buffer.buffer(body));
+            } else {
+                response.setStatusCode(204);
+                response.end();
+            }
+        });
+    }
+
     @Override
     public void handle(RoutingContext routingContext) {
-        String mediaType = routingContext.request().getHeader("Content-Type");
-        if (mediaType == null || mediaType.startsWith("application/json") || mediaType.trim().equals("")) {
-            if (routingContext.request().getHeader("ce-id") != null) {
-                binaryContentMode(routingContext);
-            } else {
-                regularFunqyHttp(routingContext);
+        final HttpServerRequest request = routingContext.request();
+        final String mediaType = request.getHeader("Content-Type");
+        boolean binaryCE = request.headers().contains("Ce-Id");
+        boolean structuredCE = false;
+        if (mediaType != null) {
+            structuredCE = mediaType.startsWith("application/cloudevents+");
+        }
+
+        if (structuredCE || binaryCE) {
```

Comment:
Why this code to distinguish between binary and structured? Can't you just try to create a message reader and if it fails, because the input is not a cloudevent, you handle it with the other logic you want to execute?